manual/src/main/kotlin/com.r3.corda.finance.manual/flows/UpdatePaymentStatusManually.kt

```diff
@@ -0,0 +1,105 @@
+package com.r3.corda.finance.manual.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.finance.obligation.client.getLinearStateById
+import com.r3.corda.finance.obligation.client.resolver
+import com.r3.corda.finance.obligation.commands.ObligationCommands
+import com.r3.corda.finance.obligation.contracts.ObligationContract
+import com.r3.corda.finance.obligation.states.Obligation
+import com.r3.corda.finance.obligation.types.PaymentReference
+import com.r3.corda.finance.obligation.types.PaymentStatus
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.flows.*
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.transactions.WireTransaction
+import net.corda.core.utilities.ProgressTracker
+
+object UpdatePaymentStatusManually {
+
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(
+            private val paymentReference: PaymentReference,
+            private val status: PaymentStatus,
+            private val linearId: UniqueIdentifier
+    ) : FlowLogic<WireTransaction>() {
+
+        companion object {
+            object INITIALISING : ProgressTracker.Step("Performing initial steps.")
+            object BUILDING : ProgressTracker.Step("Building and verifying transaction.")
+            object SIGNING : ProgressTracker.Step("signing transaction.")
+
+            object COLLECTING : ProgressTracker.Step("Collecting counterparty signature.") {
+                override fun childProgressTracker() = CollectSignaturesFlow.tracker()
+            }
+
+            object FINALISING : ProgressTracker.Step("Finalising transaction.") {
+                override fun childProgressTracker() = FinalityFlow.tracker()
+            }
+
+            fun tracker() = ProgressTracker(INITIALISING, BUILDING, SIGNING, COLLECTING, FINALISING)
+        }
+
+        override val progressTracker: ProgressTracker = tracker()
+
+        @Suspendable
+        override fun call(): WireTransaction {
+            progressTracker.currentStep = INITIALISING
+            val obligationStateAndRef = getLinearStateById<Obligation<*>>(linearId, serviceHub)
+                    ?: throw IllegalArgumentException("LinearId not recognised.")
+            val obligationState = obligationStateAndRef.state.data
+
+            val os = obligationState.withWellKnownIdentities(resolver)
+            val obligee = os.obligee
+            check(ourIdentity == obligee) { "This flow can only be started by the obligee. " }
+            val obligor = serviceHub.identityService.requireWellKnownPartyFromAnonymous(os.obligor)
+            val payment = obligationState.payments.find { it.paymentReference == paymentReference }
+            check(payment != null) { "Could not find payment with reference '$paymentReference'" }
```

Comment:
you can use `if` here to get the smart cast, so no need to subsequently use unsafe dereferencing. No need to change anything though!